iocage_lib/iocage.py

```diff
@@ -2160,8 +2186,44 @@ def debug(self, directory):
 
         ioc_debug.IOCDebug(directory).run_debug()
 
-    def snap_remove(self, snapshot):
+    def _get_cloned_datasets(self):
+        return {
+                    d.properties.get('origin', "").replace('/root@', '@')
+                    for d in Dataset(
+                        os.path.join(self.pool, 'iocage')
+                    ).get_dependents(depth=3)
```

Comment:
Is there a reason for limiting depth to 3 here?

Reply:
Yes that will give us up to the `/root` dataset but nothing below. (As in `/iocage/jails/jailname/root`.)

Nothing is created below by iocage so the only case there would be something is if the user manually created children datasets.

Anyway we do not care because here we are making a list of snapshots we want to protect from deletion and the snapshots managed by iocage (which could be deleted) are only on the jail/template level or its subsequent `/root`.